Fix mistakes in the usage examples

Avaq opened the pull request:
Closes #71 

/cc @kasselTrankos

Avaq committed:
Fix mistakes in the usage examples

Avaq commented on index.js:
```diff
@@ -21,7 +21,10 @@
 //. import {resolve} from 'fluture/index.js';
 //.
-//. const S = sanctuary.create ({checkTypes: true, env: S.env.concat (env)});
+//. const S = sanctuary.create ({
+//.   checkTypes: true,
+//.   env: sanctuary.env.concat (env),
+//. });
```
I think that for many new users, this may be the first time they are introduced to `sanctuary.create`. For that reason, it might be nice to add a comment by `checkTypes: true` that they need to configure it such that type checking is disabled in production. What do you think @davidchambers ?

Avaq replied:
Or maybe I should just put `process.env.NODE_ENV !== 'production'` there instead of `true`.

davidchambers replied:
> Or maybe I should just put `process.env.NODE_ENV !== 'production'` there instead of `true`.

Good idea.

davidchambers commented on index.js:
```diff
@@ -21,7 +21,10 @@
-//. const S = sanctuary.create ({checkTypes: true, env: S.env.concat (env)});
+//. const S = sanctuary.create ({
+//.   checkTypes: true,
+//.   env: sanctuary.env.concat (env)
```
Trailing comma?

Avaq replied:
I had it there before, https://github.com/fluture-js/fluture-sanctuary-types/compare/18e393167a21c2a66a4a4699e6f9b64ed803882d..2b45ef56ffa6a54d96acbbfa4125bf0f89ff0e6f, but my linter doesn't like it: https://github.com/fluture-js/fluture-sanctuary-types/actions/runs/485462409

Avaq replied:
I should update the linting rules in https://github.com/fluture-js/fluture-template some day.

Avaq replied:
Ah, I think it uses the wrong linting rules. The source is es3 compatible, but the docs don't have to be.

Avaq committed:
Update README examples to recommend better Sanctuary configs

In particular, don't recommend blindly turning on
type checking in every environment.
